src/routes/safe/components/Apps/components/AppCard/index.tsx

```diff
@@ -125,15 +146,37 @@ const AppCard = ({ app, iconSize = 'md', to, onPin, onRemove, pinned = false }:
     )
   }
 
+  const shareSafeAppLabel = `Share ${app.name} Safe App`
+
   const content = (
     <>
+      <LeftIconBtn
+        aria-label={shareSafeAppLabel}
+        title={shareSafeAppLabel}
+        onClick={(e) => {
+          // prevent triggering the link event
+          e.preventDefault()
+
+          const baseUrl = `${window.location.origin}${PUBLIC_URL}`
+          const shareSafeAppUrl = `${baseUrl}${SAFE_APP_LANDING_PAGE_ROUTE}?appUrl=${encodeURI(
```

Comment:
It would be good to move this to a util function in case we want to reuse it somewhere

Reply:
I would also extract the logic from everywhere else that generates Safe Apps links.